build-tools/src/main/java/org/elasticsearch/gradle/testclusters/TestClustersAware.java

```diff
@@ -34,11 +35,17 @@ default void useCluster(ElasticsearchCluster cluster) {
         if (cluster.getPath().equals(getProject().getPath()) == false) {
             throw new TestClustersException("Task " + getPath() + " can't use test cluster from" + " another project " + cluster);
         }
-
-        cluster.getNodes()
-            .all(node -> node.getDistributions().forEach(distro -> dependsOn(getProject().provider(() -> distro.maybeFreeze()))));
-        dependsOn(cluster.getPluginAndModuleConfigurations());
+        if (cluster.getName().equals(getName())) {
+            for (ElasticsearchNode node : cluster.getNodes()) {
+                for (ElasticsearchDistribution distro : node.getDistributions()) {
+                    ElasticsearchDistribution frozenDistro = distro.maybeFreeze();
+                    dependsOn(frozenDistro);
+                }
+            }
+            dependsOn(cluster.getPluginAndModuleConfigurations());
+        }
```

Comment:
`numberOfNodes` is still used across the build in the build scripts. idea is bad in finding those references automatically for you in build scripts unfortunately (especially with the gradle dsl).
The numberOfNodes property should  not be changed after we have triggered the useCluster here. I tweaked this to fail in case this does happen somehow. 
Using a domain object container for ElasticSearchCluster#nodes was probably not required in the first place. It just had some convenience to rely on less configuration ordering in the TestCluster configurations across the build